doc/source/development/debugging_extensions.rst

```diff
@@ -14,14 +14,31 @@ For Python developers with limited or no C/C++ experience this can seem a daunti
   2. `Fundamental Python Debugging Part 2 - Python Extensions <https://willayd.com/fundamental-python-debugging-part-2-python-extensions.html>`_
   3. `Fundamental Python Debugging Part 3 - Cython Extensions <https://willayd.com/fundamental-python-debugging-part-3-cython-extensions.html>`_
 
-Generating debug builds
------------------------
+Using Docker
+------------
 
-By default building pandas from source will generate a release build. To generate a development build you can type::
+To simplify the debugging process, pandas has created a Docker image with a debug build of Python and the gdb/Cython debuggers pre-installed. You may either ``docker pull pandas/pandas-debug`` to get access to this image or build it from the ``tooling/debug`` folder locallly.
 
-    pip install -ve . --no-build-isolation --config-settings=builddir="debug" --config-settings=setup-args="-Dbuildtype=debug"
+You can then mount your pandas repository into this image via:
 
-By specifying ``builddir="debug"`` all of the targets will be built and placed in the debug directory relative to the project root. This helps to keep your debug and release artifacts separate; you are of course able to choose a different directory name or omit altogether if you do not care to separate build types.
+.. code-block:: sh
+
+   docker run --rm -it -w /data -v ${PWD}:/data pandas/pandas-debug
+
+Inside the image, you can use meson to build/install pandas and place the build artifacts into a ``debug`` folder using a command as follows:
+
+.. code-block:: sh
+
+    python -m pip install -ve . --no-build-isolation --config-settings=builddir="debug" --config-settings=setup-args="-Dbuildtype=debug"
+
+If planning to use cygdb, the files required by that application are placed within the build folder. So you have to first ``cd`` to the build folder, then start that application.
```

Comment:
I think this is a bug in Cython, which still uses optparse (deprecated in 3.2) insted of argparse. Seems like the tool should allow you to specify the build folder in theory, but in practice doesn't work. Will push up a PR separately to Cython to fix

Reply:
See https://github.com/cython/cython/pull/5499